network/packet_listener_handler.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/Jigsaw-Code/outline-internal-sdk/transport"
+)
+
+const packetMaxSize = 2048
+
+// Compilation guard against interface implementation
+var _ PacketHandler = (*packetListenerHandlerAdapter)(nil)
+var _ PacketSession = (*packetListenerSessionAdapter)(nil)
+
+type packetListenerHandlerAdapter struct {
+	transport.PacketListener
+}
+
+type packetListenerSessionAdapter struct {
+	net.PacketConn
+	response     PacketResponseWriter
```

Comment:
This will increase the readability. `response` is ambiguous, so is `writer`.

```suggestion
	responseWriter     PacketResponseWriter
```